### ISSUE TITLE
Sampling from MvNormal with integer arrays fails

### ISSUE THREAD

**Issue by baggepinnen** [JuliaStats/Distributions.jl#1004]
When an `MvNormal` is created using integer arrays, sampling from the distribution fails.
```julia
julia> using Distributions

julia> rand(MvNormal([1,1],[1,1]), 10)
ERROR: MethodError: no method matching randn(::Random._GLOBAL_RNG, ::Type{Int64})
Closest candidates are:
  randn(::AbstractRNG, ::Type{T}, ::Tuple{Vararg{Int64,N}} where N) where T at /home/fredrikb/julia-1.3.0-rc4/share/julia/stdlib/v1.3/Random/src/normal.jl:181
  randn(::AbstractRNG, ::Type{T}, ::Integer, ::Integer...) where T at /home/fredrikb/julia-1.3.0-rc4/share/julia/stdlib/v1.3/Random/src/normal.jl:184
  randn(::AbstractRNG) at /home/fredrikb/julia-1.3.0-rc4/share/julia/stdlib/v1.3/Random/src/normal.jl:39
  ...
Stacktrace:
 [1] randn!(::Random._GLOBAL_RNG, ::Array{Int64,2}) at /home/fredrikb/julia-1.3.0-rc4/share/julia/stdlib/v1.3/Random/src/normal.jl:173
 [2] _rand! at /home/fredrikb/.julia/packages/Distributions/2i7UF/src/multivariate/mvnormal.jl:269 [inlined]
 [3] rand at /home/fredrikb/.julia/packages/Distributions/2i7UF/src/multivariates.jl:77 [inlined]
 [4] rand(::MvNormal{Int64,PDMats.PDiagMat{Int64,Array{Int64,1}},Array{Int64,1}}, ::Int64) at /home/fredrikb/.julia/packages/Distributions/2i7UF/src/multivariates.jl:76
 [5] top-level scope at REPL[10]:1

(v1.3) pkg> st Distributions
    Status `~/.julia/environments/v1.3/Project.toml`
  [31c24e10] Distributions v0.21.5
  [1fd47b50] QuadGK v2.0.3
  [2913bbd2] StatsBase v0.32.0

julia> VERSION
v"1.3.0-rc4.1"
```

**Comment by matbesancon** [comment]
we could add a conversion method, PR welcome :)

**Comment by ben-schulz** [comment]
it looks like something similar was marked resolved way back in issue #498 -- but it would appear that change only addressed a type error on invocation of the constructor.

the code above fails because because:

```
dist = MvNormal([1,1],[1,1])
dist::MvNormal{Int32,PDMats.PDiagMat{Int32,Array{Int32,1}},Array{Int32,1}}
```

i.e., `MvNormal` successfully instantiates with element type `<: Integer`, where as an invocation to `rand` expects a float of some kind.

should the `MvNormal` constructor always promote the underlying element type to be usable with `rand`? that might make sense (seeing as it is a distribution), but it would seem to entail parameterizing `MvNormal{T}` by a type more restrictive than the current `T<:Real`

**Comment by JobJob** [comment]
Related - MvNormal with Array{Int} covariance also gives an unintuitive error:
```julia
julia> MvNormal([0, 0.0], [1 0; 0 1])
ERROR: MethodError: no method matching PDMats.PDMat{Int64,Array{Int64,2}}(::Int64, ::Array{Int64,2}, ::LinearAlgebra.Cholesky{Float64,Array{Float64,2}})
```

**Comment by mossr** [comment]
Note that the fix for this (which originates in PDMats.jl) was merged and is now part of PDMats v0.10.1

**Comment by matbesancon** [comment]
so should this be closed?

**Comment by mossr** [comment]
The fix is part of a PDMats patch release (v0.10.1), but a fresh install of Distributions grabs v0.10 (I just tested this with Julia v1.5)—so it's up to you how to proceed. I doubt putting an explicit [compat] requirement for PDMats v0.10.1 is what you'd want.

**Comment by mossr** [comment]
But my suggestion would be to—in fact—add the [compat] for "0.10.1" (which would capture [0.10.1, 0.11)
